src/transformers/models/bloom/convert_bloom_original_checkpoint_to_pytorch.py

```diff
@@ -0,0 +1,251 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Convert BigScience BLOOM checkpoint."""
+
+
+import argparse
+import json
+import os
+import re
+
+import torch
+
+from transformers import BloomConfig, BloomModel
+from transformers.file_utils import CONFIG_NAME, WEIGHTS_NAME
+from transformers.utils import logging
+
+
+logging.set_verbosity_info()
+
+WEIGHTS_TO_AVERAGE_ENDSWITH = [
+    "word_embeddings_layernorm.weight",
+    "word_embeddings_layernorm.bias",
+    "input_layernorm.weight",
+    "input_layernorm.bias",
+    "post_attention_layernorm.weight",
+    "post_attention_layernorm.bias",
+    "self_attention.dense.bias",
+    "mlp.dense_4h_to_h.bias",
+    "ln_f.weight",
+    "ln_f.bias",
+]
+
+WEIGHTS_WITH_ROW_PARALLELISM_CONTAIN = [
+    "dense_4h_to_h",
+    "self_attention.dense.weight",
+]
+
+
+def layer_name_mapping(key, file):
+    """Convert Megatron-DeepSpeed TP/PP weights mapping in transformers PP only"""
+    # Handle first and last layers
+    if key == "word_embeddings.weight":
+        return key
+    if key == "word_embeddings.norm.weight":
+        return "word_embeddings_layernorm.weight"
+    if key == "word_embeddings.norm.bias":
+        return "word_embeddings_layernorm.bias"
+    if key == "weight":
+        return "ln_f.weight"
+    if key == "bias":
+        return "ln_f.bias"
```

Comment:
5 identical if calls in a row surely call for a remapping dict, 

```suggestion
    layer_name_map = {
        "word_embeddings.weight": "word_embeddings.weight",
        "word_embeddings.norm.weight":"word_embeddings_layernorm.weight",
        "word_embeddings.norm.bias":"word_embeddings_layernorm.bias",
        "weight":"ln_f.weight",
        "bias":"ln_f.bias",
    }

    if key in layer_name_map:
        return layer_name_map[key]        
```

(untested)

Reply:
may be also moving up to where all the other maps are?